File: AbstractFactory/cpp-source/rooms/CavesNormalRoom.cpp
//
// Created by Pavel Akhtyamov on 2019-02-25.
//

#include "CavesNormalRoom.h"

std::string CavesNormalRoom::GenerateBoss() const {
  return "300 hp";
}

std::string CavesNormalRoom::GenerateLoot() const {
  return "Loot x3";
}

size_t CavesNormalRoom::chance() const {
  return 2;
}

File: JavaSource/src/main/java/ru/akhcheck/patterns/abstractfactory/rooms/CavesNormalRoom.java
package ru.akhcheck.patterns.abstractfactory.rooms;

/** Обычная комната пещер. */
public class CavesNormalRoom implements NormalRoom {
    @Override
    public String generateLoot() {
        return "Caves loot";
    }

    @Override
    public String generateBoss() {
        return "Chub";
    }
}
